src/Development/IDE/Plugin/Completions/Logic.hs

```diff
@@ -374,9 +374,28 @@ toggleSnippets ClientCapabilities { _textDocument } (WithSnippets with) x
 getCompletions :: IdeOptions -> CachedCompletions -> ParsedModule -> VFS.PosPrefixInfo -> ClientCapabilities -> WithSnippets -> IO [CompletionItem]
 getCompletions ideOpts CC { allModNamesAsNS, unqualCompls, qualCompls, importableModules }
                pm prefixInfo caps withSnippets = do
-  let VFS.PosPrefixInfo { VFS.fullLine, VFS.prefixModule, VFS.prefixText } = prefixInfo
-      enteredQual = if T.null prefixModule then "" else prefixModule <> "."
-      fullPrefix  = enteredQual <> prefixText
+  let VFS.PosPrefixInfo { VFS.fullLine, VFS.prefixModule, VFS.prefixText, VFS.cursorPos } = prefixInfo
+      maybeIndex ix = T.take 1 . T.drop ix
+
+      typedDot = "." == maybeIndex (_character cursorPos) fullLine
+
+      qualifiedBit
+        | T.null prefixModule = ""
+        | otherwise           = prefixModule <> "."
+
+      {- Items are only added to prefixModule after the first letter after a dot is typed.
+         "import Control"             -> CC { prefixModule = ""       , prefixText = "Control"    }
+         "import Control."            -> CC { prefixModule = ""       , prefixText = "Control"    }
+         "import Control.C"           -> CC { prefixModule = "Control", prefixText = "C"          }
+         "import Control.Concurrent"  -> CC { prefixModule = "Control", prefixText = "Concurrent" }
+         "import Control.Concurrent." -> CC { prefixModule = "Control", prefixText = "Concurrent" }
```

Comment:
Have you considered raising this in haskell-lsp? I cannot think a reason for this behaviour
@alanz do you remember if this behaviour is by design?

Reply:
@pepeiborra I am not the person who did most of the completion stuff. Best to look in the hie changelog, I suspect.

Reply:
that's all right, @lazamar tracked the bug back to ghcide (stale data)